provider/cmd/hostname_operator_test.go

```diff
@@ -156,14 +193,19 @@ func TestHostnameOperatorApplyDelete(t *testing.T) {
 
 	_, exists := s.op.hostnames[hostname]
 	require.False(t, exists) // Removed from dataset
+
+	require.True(t, s.op.hostnamesData.needsPrepare)
+	require.NoError(t, s.op.prepare())
+	require.False(t, s.op.hostnamesData.needsPrepare)
+
 }
 
 func TestHostnameOperatorApplyDeleteFails(t *testing.T) {
 	s := makeHostnameOperatorScaffold(t)
 	require.NotNil(t, s)
 	defer s.cancel()
 
-	const hostname = "qux.io"
+	const hostname = "lsn.io"
```

Comment:
Let's use [`.test` TLD](https://datatracker.ietf.org/doc/html/rfc2606#section-2).  There was some drama/hilarity recently with a test suite that was spamming a site that I can't find right now.